### PR TITLE
TextEntryDialog: Rewrite without UI files

### PR DESCRIPTION
**Pull request by danrabbit**
Rewrites the dialog with GObject-style construction (almost) and no UI files or Gtk.Builder or anything.

Solves a few deprecation warnings

### CODE REVIEW

File: src/Dialogs/TextEntryDialog.vala [hunk 1]
```diff
@@ -19,51 +19,62 @@
...
-    private Gtk.Button button1;
-    private Gtk.Button button2;
-    private Gtk.ButtonBox action_area_box;
 
-    public void set_builder (Gtk.Builder builder) {
-        this.builder = builder;
-    }
+    public TextEntryDialog (OnModifyValidateType? modify_validate, string title, string label,
+string? initial_text, Gee.Collection<string>? completion_list, string? completion_delimiter) {
+        Object (
```
Comment (jeremypw) [inline src/Dialogs/TextEntryDialog.vala]: Align parameters under opening bracket?

File: src/Dialogs/TextEntryDialog.vala [hunk 2]
```diff
@@ -19,51 +19,62 @@
...
+    construct {
+        var name_label = new Gtk.Label (label);
+        name_label.halign = Gtk.Align.START;
+        name_label.get_style_context ().add_class (Granite.STYLE_CLASS_H4_LABEL);
 
-        entry = builder.get_object ("entry") as Gtk.Entry;
-        entry.set_text (initial_text != null ? initial_text : "");
+        entry = new Gtk.Entry ();
+        entry.hexpand = true;
+        entry.text = initial_text != null ? initial_text : "";
```
Comment (jeremypw) [inline src/Dialogs/TextEntryDialog.vala]: `entry.text = initial_text ?? ""` is more concise.

**Comment by jeremypw** [comment]
Note: the alignment example did not work in the review comment - not WYSIWYG :(

**Comment by danrabbit** [comment]
Yeah I think we should probably document an official style for how to deal with this.

My reasoning for using this style is largely based around when we started using linters in other projects and having a single indent for things between brackets on multiple lines is what the linter expects.

I also feel like it's more maintainable if the parameters change. You don't necessarily have to touch a bunch of lines to fix the alignment.

**Pull request merged**